lib/developer_portal/config/initializers/liquid_partial_cache.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+# Patch Liquid::PartialCache to support context-aware file systems
+# This is needed because Liquid 5.x removed context parameter from FileSystem#read_template_file
+# (see https://github.com/Shopify/liquid/pull/441),
+# but our CMS::DatabaseFileSystem needs access to context.registers[:draft]
+module Liquid
+  class PartialCache
+    class << self
+      alias_method :load_without_context_injection, :load
+
+      def load(template_name, context:, parse_context:)
+        file_system = context.registers[:file_system]
+
+        # Inject context into file system if it supports it
+        file_system.current_context = context if file_system.respond_to?(:current_context=)
```

Comment:
If we are using the context only to retrieve the draft, wouldn't it be better to just inject the draft? Maybe the draft can be placed somewhere else to avoid the monkey patching? Just asking, I have no idea.

Also, the PR linked above mentions the context is removed to better manage some cache. Doesn't this context re-injection interfere in that caching?